packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_typedefs.hpp

```diff
@@ -71,18 +72,18 @@ using fe_multivector_type = Tpetra::FEMultiVector<Scalar>;
 
 
 using global_ordinal_view_type =
-  Kokkos::View<global_ordinal_type*, execution_space>;
+  Kokkos::View<global_ordinal_type*, device_type>;
```

Comment:
What determines whether to use execution_space or device_type?

Reply:
Here, this needs to be device_type in general so that it gets both execution space and memory space from the Node type. In the case of HIP, this means it's allocated in HIPHostPinnedSpace so that it's host accessible. Important since this test still relies on UVM. It also means that when we don't need UVM anymore, there's only one place to switch to HIPSpace (the KokkosHIPWrapperNode typedef).